### PR TITLE
feat(sagemaker): network isolation for a model

### PR DESCRIPTION
**Pull request by badmintoncryer**
### Issue # (if applicable)

None

### Reason for this change

SageMaker model supports for the [network isolation](https://docs.aws.amazon.com/sagemaker/latest/dg/mkt-algo-model-internet-free.html) function. But Model construct does not support this.

### Description of changes

Add `networkIsolation` to `ModelProps`

```ts
const model = new sagemaker.Model(this, 'ContainerModel', {
  containers: [
    {
      image,
      modelData,
    }
  ],
  networkIsolation: true, // Added
});
```

### Description of how you validated changes

Added both unit and integ tests

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-sagemaker-alpha/lib/model.ts [hunk 1]
```diff
@@ -206,6 +206,17 @@ export interface ModelProps {
...
   readonly allowAllOutbound?: boolean;
+
+  /**
+   * Whether to isolate the model container.
+   *
+   * No inbound or outbound network calls can be made to or from the model container.
+   *
+   * @see https://docs.aws.amazon.com/sagemaker/latest/dg/mkt-algo-model-internet-free.html
+   *
+   * @default - false
```
Comment (msambol) [inline packages/@aws-cdk/aws-sagemaker-alpha/lib/model.ts:217]: no hyphen here

**Review by msambol — APPROVED** [review]
One small nit, otherwise LGTM.

**Comment by badmintoncryer** [comment]
@msambol Thanks always! I've fixed comments.

File: packages/@aws-cdk/aws-sagemaker-alpha/lib/model.ts [hunk 2]
```diff
@@ -206,6 +206,17 @@ export interface ModelProps {
    * @default true
    */
   readonly allowAllOutbound?: boolean;
+
+  /**
+   * Whether to isolate the model container.
```
Comment (msambol) [inline packages/@aws-cdk/aws-sagemaker-alpha/lib/model.ts:211]: I think this could be more descriptive. Maybe combine this with the sentence below.

Comment (badmintoncryer) [reply packages/@aws-cdk/aws-sagemaker-alpha/lib/model.ts]: I have combined the sentences below. If you have any better suggestions, please let me know!

File: packages/@aws-cdk/aws-sagemaker-alpha/lib/model.ts [hunk 3]
```diff
@@ -206,6 +206,16 @@ export interface ModelProps {
    * @default true
    */
   readonly allowAllOutbound?: boolean;
+
+  /**
+   * Whether to isolate the model container,
+   * ensuring that no inbound or outbound network calls can be made to or from the model container.
```
Comment (msambol) [inline packages/@aws-cdk/aws-sagemaker-alpha/lib/model.ts:212]:
```
Whether to isolate to enable network isolation for the model container.

When enabled, no inbound or outbound network calls can be made to or from the model container.
```

File: packages/@aws-cdk/aws-sagemaker-alpha/lib/model.ts [hunk 4]
```diff
@@ -206,6 +206,17 @@ export interface ModelProps {
    * @default true
    */
   readonly allowAllOutbound?: boolean;
+
+  /**
+   * Whether to isolate to enable network isolation for the model container.
```
Comment (msambol) [inline packages/@aws-cdk/aws-sagemaker-alpha/lib/model.ts:211]: I messed up, sorry: `Whether to enable network isolation for the model container.`

File: packages/@aws-cdk/aws-sagemaker-alpha/lib/model.ts (hunk 3 above)
Comment (badmintoncryer) [reply packages/@aws-cdk/aws-sagemaker-alpha/lib/model.ts]: Thanks!

**Review by msambol — APPROVED** [review]
LGTM!

**Review by shikha372 — DISMISSED** [review]

**Comment by badmintoncryer** [comment]
@mergify update

**Comment by badmintoncryer** [comment]
@shikha372 Thank you for your approval. Could you please resolve the mergify job problem?

**Review by shikha372 — APPROVED** [review]

**Pull request merged**